pkg/spiffe/spiffe.go

```diff
@@ -0,0 +1,59 @@
+package spiffe
+
+import (
+	"fmt"
+	"os"
+)
+
+const (
+	Scheme = "spiffe"
+
+	// The default SPIFFE URL value for trust domain
+	defaultTrustDomain = "cluster.local"
+)
+
+var trustDomain = defaultTrustDomain
+
+func SetTrustDomain(value string) {
+	trustDomain = value
+}
+
+func GetTrustDomain() string {
+	return trustDomain
+}
+
+func DetermineTrustDomain(commandLineTrustDomain string, domain string, isKubernetes bool) string {
+
+	if len(commandLineTrustDomain) != 0 {
+		return commandLineTrustDomain
+	}
+	envTrustDomain := os.Getenv("ISTIO_SA_DOMAIN_CANONICAL")
```

Comment:
Can you add any environment used in pilot or agent to features/pilot ? With a lot of comments please - the name makes no sense to someone who is not familiar with spiffe, and seems pretty generic - if it is only spiffee please name it accordingly..

Reply:
We removed the use of the environment variable.